### PR TITLE
User/athanaseus/fill properties into tango db

### PR DESCRIPTION
**Pull request by Athanaseus**

### CODE REVIEW

File: tango_simlib/tango_sim_generator.py [hunk 1]
```diff
@@ -271,6 +275,13 @@ def _reset_to_default_state(self):
                         adjustable_val = 0.0
                     setattr(simulated_quantity, attr, adjustable_val)
 
+        def fill_new_device_properties(self):
```
Comment (Athanaseus) [inline tango_simlib/tango_sim_generator.py]: _write

File: tango_simlib/tango_sim_generator.py [hunk 2]
```diff
@@ -271,6 +275,13 @@ def _reset_to_default_state(self):
...
+            """Fill device properties into tangoDB"""
+            db_instance = Database()
+            for prop_name, prop_meta in self.model.sim_properties.items():
+                db_instance.put_device_property(
+                    self.get_name(), {prop_name: prop_meta['DefaultPropValue']})
+
```
Comment (lvdheever) [inline tango_simlib/tango_sim_generator.py]: Are you sure there are no other meta_data on Properties that should be written to the DB?

File: tango_simlib/model.py [hunk 3]
```diff
@@ -529,6 +533,45 @@ def action_handler(model, data_input=None, tango_dev=None):
...
+
+    """
+    def __init__(self, parser_instance, tango_device_name, sim_model=None):
+        self.parser_instance = parser_instance
+        if sim_model:
+            if isinstance(sim_model, Model):
+                self.sim_model = sim_model
+            else:
+                raise SimModelException("The sim_model object passed is not an "
+                    "instance of the class mkat_tango.simlib.model.Model")
```
Comment (ajoubertza) [inline tango_simlib/model.py]: Please fix the indentation here. This should line up with the start of the string on the previous line.

File: tango_simlib/tests/test_tango_sim_generator.py [hunk 4]
```diff
@@ -95,6 +96,18 @@ def test_device_command_list(self):
                           "The commands specified in the xmi file are not present in"
                           " the device")
 
+    def test_device_property_list(self):
+        """Testing whether the device properties in the model are added to
+        the TANGO sim device as expected
+        """
+        db = PyTango.Database()
```
Comment (ajoubertza) [inline tango_simlib/tests/test_tango_sim_generator.py]: I think you should mock this database object.  You don't want the tests to try to connect to a real database.  Also when you start up this test device, it is running in the nodb mode.  Best we can do is check that `put_device_property` was called on the mock with correct parameters.

Comment (ajoubertza) [inline tango_simlib/tests/test_tango_sim_generator.py]: @Athanaseus and I had a look at this.  It won't be possible to mock, as the device server is running in a different process to the tests.  Suggest this test is removed.

We also realised that the device server shouldn't try and access the database if launched in the "nodb" mode, like we do for the tests.

**Comment by Athanaseus** [comment]
Developments for LMC-360

**Comment by Athanaseus** [comment]
In[1]: from PyTango import DataBase
In[2]: db_instance = DataBase()
In[3]: db_instance.get_device_property_list('test/weather/1', '*')
Out[4]: DbDatum(name = 'test/weather/1', value_string = ['simulated_values', 'sim_xmi_description_file'])

In [5]: db_instance.get_info()
Out[5]: 'TANGO Database sys/database/2\n \nRunning since 2017-07-28 09:05:36\n \nDevices defined  = 65\nDevices exported  = 14\nDevice servers defined  = 32\nDevice servers exported  = 6\n \nDevice properties defined  = 98 [History lgth = 442]\nClass properties defined  = 86 [History lgth = 165]\nDevice attribute properties defined  = 14 [History lgth = 35]\nClass attribute properties defined  = 0 [History lgth = 0]\nObject properties defined  = 0 [History lgth = 0]\n'

In[1]: from PyTango import DataBase
In[2]: db_instance = DataBase('weather_ds_tango.db')
In[3]: self.db_instance.get_device_property_list('test/nodb/tangodeviceserver#dbase=no', '*')
omniORB: ERROR -- the application attempted to invoke an operation on a nil reference.
---------------------------------------------------------------------------
RuntimeError                              Traceback (most recent call last)
/home/athanaseus/Documents/SKA_Software_Engeneering/git@SKA/tango-simlib/tango_simlib/tests/test_tango_sim_generator.py in <module>()
----> 1 self.db_instance.get_device_property_list('test/nodb/tangodeviceserver', '*')

/usr/local/lib/python2.7/dist-packages/tango/db.pyc in __Database__get_device_property_list(self, dev_name, wildcard, array)
    436             Throws     : ConnectionFailed, CommunicationFailed, DevFailed from device"""
    437     if array is None:
--> 438         return self._get_device_property_list(dev_name, wildcard)
    439     elif isinstance(array, StdStringVector):
    440         return self._get_device_property_list(dev_name, wildcard, array)

RuntimeError: unidentifiable C++ exception


In[4]: self.db_instance.get_info()
Out[4]: 'TANGO FileDatabase  /tmp/tmpR_MXJP/weather_ds_tango.db\n\nRunning since ----\n\nDevices defined = 2\nDevices exported = 2\nDevice servers defined = 1\nDevice servers exported = 1\n\nClass properties defined = 0\nDevice properties defined = 2\nClass attribute properties defined = 0\nDevice attribute properties defined = 0\n'

**Comment by Athanaseus** [comment]
Was able to obtain database info and get properties count. For some reason it fails when we query for a list. More investigation will follow with LMC-309 (https://skaafrica.atlassian.net/browse/LMC-309) considering that our testing class (https://github.com/vxgmichel/pytango-devicetest) is deprecated

**Pull request merged**